pennylane/transforms/decompositions/clifford_plus_t/cliffordt_transform.py

```diff
@@ -0,0 +1,368 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transform function for the Clifford+T decomposition."""
+
+import math
+from itertools import product
+from typing import Sequence, Callable
+import pennylane as qml
+from pennylane.transforms.core import transform
+from pennylane.tape import QuantumTape
+from pennylane.transforms.optimization import (
+    cancel_inverses,
+    commute_controlled,
+    merge_rotations,
+    remove_barrier,
+)
+from pennylane.transforms.optimization.optimization_utils import find_next_gate, _fuse_global_phases
+
+# Single qubits Clifford+T gates in PL
+_CLIFFORD_T_ONE_GATES = [
+    qml.Identity,
+    qml.PauliX,
+    qml.PauliY,
+    qml.PauliZ,
+    qml.Hadamard,
+    qml.S,
+    qml.SX,
+    qml.T,
+]
+
+# Two qubits Clifford+T gates in PL
+_CLIFFORD_T_TWO_GATES = [
+    qml.CNOT,
+    qml.CY,
+    qml.CZ,
+    qml.SWAP,
+    qml.ISWAP,
+]
+
+_PARAMETER_GATES = [qml.RX, qml.RY, qml.RZ, qml.Rot, qml.GlobalPhase]
+
+_CLIFFORD_T_GATES = _CLIFFORD_T_ONE_GATES + _CLIFFORD_T_TWO_GATES
+
+
+def check_clifford_op(op):
+    r"""Check if an operator is Clifford or not.
+
+    For a given unitary operator :math:`U` acting on :math:`N` qubits, this method checks that the
+    transformation :math:`UPU^{\dagger}` maps the Pauli tensor products :math:`P = {I, X, Y, Z}^{\otimes N}`
+    to Pauli tensor products with :math:`O(N \times 8^N)` time complexity when using naive matrix multiplication.
+
+    Args:
+        op: the operator that needs to be tested
+
+    Returns:
+        Bool that represents whether the provided operator is Clifford or not.
+    """
+
+    # Check if matrix can be calculated for the operator
+    if not op.has_matrix:
+        return False
+
+    # Compute the LCUs for the operator in Pauli basis
+    num_qubits = len(op.wires)
+    op = qml.map_wires(op, wire_map={wire: idx for idx, wire in enumerate(op.wires)})
```

Comment:
Why do we need to map the wires? I can't find a usage of `op` other than in calculating the matrix.

Reply:
I was doing this to ensure that using different wire labels doesn't somehow give me a different matrix.

Reply:
`qml.matrix(op)` should always conform to the wire order specified by `op.wires`, so I'm not sure we need to do so in this case.